tt_metal/llrt/watcher.cpp

```diff
@@ -495,7 +498,13 @@ static void watcher_loop(int sleep_usecs) {
                 fprintf(logfile, "No active devices\n");
             }
 
-            dump(logfile, false);
+            try {
+                dump(logfile, false);
+            } catch (std::runtime_error& e) {
+                watcher::watcher_killed_due_to_error = true;
+                watcher::enabled = false;
+                break;
+            }
```

Comment:
Is the flow now that throw's only occur in wait_finish in the queue?  I don't think that'll work for all cases, it is possible a program could enqueue work for a long time w/o ever calling finish (that's not the case in our tests).

Reply:
Updated this as we discussed, I've manually verified that the exception throws instantly if the setting is disabled (which is default).